library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -31,7 +40,27 @@ public DownloadManagerBuilder withVerboseLogging() {
 
     public DownloadManager build() {
         ContentResolver contentResolver = context.getContentResolver();
-        return new DownloadManager(context, contentResolver, verboseLogging);
+        DownloadManager downloadManager = new DownloadManager(context, contentResolver, verboseLogging);
+        updateOnNetworkChanges(downloadManager);
+        return downloadManager;
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private void updateOnNetworkChanges(final DownloadManager downloadManager) {
+        if (deviceDoesNotSupportConnectivityChangesBroadcast()) {
```

Comment:
use a guard clause for easier reading + 1 less level of indentation & preferring the positive

```

if (deviceSupportsConnectivityChangesBroadcast()) {
 return;
}

ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
             NetworkRequest networkRequest = new NetworkRequest.Builder().addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET).build();
            connectivityManager.registerNetworkCallback(networkRequest, new ConnectivityManager.NetworkCallback() {
                 @Override
                 public void onAvailable(Network network) {
                     Log.v("network is active now from inside the download manager");
                     downloadManager.forceStart();
                 }
             });
```